### DISCUSSION TITLE
Minecraft Server With Custom Maps...

### DISCUSSION THREAD

**Discussion by dhruvg102** [itzg/docker-minecraft-server#2670] [Q&A] [answered]
This Question seems really dumb to ask since I can't find an answer so I assume its just very simple to do. But I am new to minecraft server hosting in general. Here's the issue:-

I am trying to get a Skyblock Minecraft server up and running so my friends and I can play together. I am using docker compose and the image from
![log](https://github.com/itzg/docker-minecraft-server/assets/71901712/c25b476b-7e97-4db2-8527-bf7da0f93f20)
 [itzg/docker-minecraft-server](https://github.com/itzg/docker-minecraft-server) which works just fine for a normal vanilla map.

But as soon I try to ass the skyblock map files from [Skyblock-1.20-Map](https://www.planetminecraft.com/project/skyblock-1-20/) it all just stops working. (Log Attached below)

I am sure its either a folder hierarchy issue or I just don't understand the yaml file or probably both so any help is appreciated.

The Folder Structure is attached as well.

P.S. I tried putting the map under a world folder but that didnt work.

The Docker Compose File: (minecraftdata is a premade docker volume.)

version: "3.8"

services:

mc:

image: itzg/minecraft-server

volumes:

- minecraftdata:/data

- /home/Minecraft/SKYBLOCK-1.20/*:/data/world

tty: true

stdin_open: true

ports:

- "25565:25565"

environment:

EULA: "TRUE"

volumes:

minecraftdata:

external: true

**Comment by itzg** [comment]
There are a couple of things wrong with volume entry

```
/home/Minecraft/SKYBLOCK-1.20/*:/data/world
```

1. You can't use a wildcard `*` in the volume mount source. Take a look [at the Docker docs](https://docs.docker.com/compose/compose-file/05-services/#volumes) for more info on volume mounts.
2. You can't mount `/data/world` as a subdirectory of an existing volume mount at `/data`

Instead make use of https://docker-minecraft-server.readthedocs.io/en/latest/misc/world-data/#cloning-world-from-a-container-path

**Comment by dhruvg102** [comment]
Hi sorry been working away at this. I was able to find [examples/docker-compose-world-download](https://github.com/itzg/docker-minecraft-server/blob/master/examples/docker-compose-world-download.yml) which seems to be perfect except using the file does not seem to make any changes. 

> Instead make use of [cloning-world-from-a-container-path]( https://docker-minecraft-server.readthedocs.io/en/latest/misc/world-data/#cloning-world-from-a-container-path)
I will next try to use this resource see where that leads me.

**Reply by itzg** [reply]
Please clarify the latest that you tried. Also be aware that it won't overwrite an existing world directory that exists, so you'll need to stop the container and remove the world directory.

**Reply by dhruvg102** [reply, marked as answer]
I have got the server working on my local network. The docker-compose.yml looks like this

version: "3.8"

services:
  mc:
    image: itzg/minecraft-server
    volumes:
      - minecraftdata:/data
    tty: true
    stdin_open: true
    ports:
      - "25565:25565"
    environment:
      EULA: "TRUE"
      VERSION: "1.20.1"
      MEMORY: "5G"

      OVERRIDE_ICON: "TRUE"
      WORLD: https://download2354.mediafire.com/k99qzyrktgfgjeke0hx1EOdmzi57zHvS2eK2If3kgoCCB8BPa45Cjf2AtszpI8VImgRAqVnM-tgAMxYZXYqxBZp4nUpkkEW7x5jIbsvHmt5zRPSNS97bQhjoEwJaBw1yt9b0qNqpiCl4HT5-GudT1X6fnjc7I9cIBNKOuXWABKWPXrg/8pxrmqijr4ry5ak/Skyblock+4.12.3.zip
      restart: unless-stopped

volumes:
  minecraftdata:
    external: true

I had to restart the container and remove the docker volume and make a new one.
